// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/bolt_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.cdp.cof";
function createBaseBoltSignals() {
    return { boltUseCase: 0, boltCdnExperimentationId: 0, boltIsContentPopular: false, boltClientAppState: 0 };
}
export const BoltSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.boltUseCase !== 0) {
            writer.uint32(8).uint32(message.boltUseCase);
        }
        if (message.boltCdnExperimentationId !== 0) {
            writer.uint32(16).uint32(message.boltCdnExperimentationId);
        }
        if (message.boltIsContentPopular === true) {
            writer.uint32(24).bool(message.boltIsContentPopular);
        }
        if (message.boltClientAppState !== 0) {
            writer.uint32(32).int32(message.boltClientAppState);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBoltSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.boltUseCase = reader.uint32();
                    break;
                case 2:
                    message.boltCdnExperimentationId = reader.uint32();
                    break;
                case 3:
                    message.boltIsContentPopular = reader.bool();
                    break;
                case 4:
                    message.boltClientAppState = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseBoltSignals();
        message.boltUseCase = (_a = object.boltUseCase) !== null && _a !== void 0 ? _a : 0;
        message.boltCdnExperimentationId = (_b = object.boltCdnExperimentationId) !== null && _b !== void 0 ? _b : 0;
        message.boltIsContentPopular = (_c = object.boltIsContentPopular) !== null && _c !== void 0 ? _c : false;
        message.boltClientAppState = (_d = object.boltClientAppState) !== null && _d !== void 0 ? _d : 0;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
