// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/creative_tools_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.cdp.cof";
function createBaseCreativeToolsSignals() {
    return { withAnimatedOverlay: false, withMusic: false };
}
export const CreativeToolsSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.withAnimatedOverlay === true) {
            writer.uint32(8).bool(message.withAnimatedOverlay);
        }
        if (message.withMusic === true) {
            writer.uint32(16).bool(message.withMusic);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreativeToolsSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.withAnimatedOverlay = reader.bool();
                    break;
                case 2:
                    message.withMusic = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseCreativeToolsSignals();
        message.withAnimatedOverlay = (_a = object.withAnimatedOverlay) !== null && _a !== void 0 ? _a : false;
        message.withMusic = (_b = object.withMusic) !== null && _b !== void 0 ? _b : false;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
