// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/feature_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
import { MediaMetadata } from "../../snapdoc/media";
import { CameraSignals } from "./camera_provided_signals";
import { ContentManagerSignals } from "./content_manager_provided_signals";
import { BoltSignals } from "./bolt_provided_signals";
import { LensesSignals } from "./lenses_provided_signals";
import { CreativeToolsSignals } from "./creative_tools_provided_signals";
import { RoutingSignals } from "./routing_provided_signals";
import { MdpMediaAttribution } from "./mdp_media_attribution";
import { MediaSignals } from "./media_provided_signals";
import { PerceptionSignals } from "./perception_provided_signals";
import { OperaSignals } from "./opera_provided_signals";
import { RecipientsSignals } from "./recipients_provided_signals";
import { BillboardSignals } from "./billboard_provided_signals";
import { CognacSignals } from "./cognac_provided_signals";
import { Value } from "../../common/value";
import { SnapKitSignals } from "./snap_kit_provided_signals";
import { UploadSignals } from "./upload_provided_signals";
import { Ruid } from "../../common/ruid";
export const protobufPackage = "snapchat.cdp.cof";
export var FeatureProvidedSignals_StoryType;
(function (FeatureProvidedSignals_StoryType) {
    FeatureProvidedSignals_StoryType[FeatureProvidedSignals_StoryType["UNKNOWN_STORY_TYPE"] = 0] = "UNKNOWN_STORY_TYPE";
    FeatureProvidedSignals_StoryType[FeatureProvidedSignals_StoryType["PUBLIC_USER_STORY"] = 1] = "PUBLIC_USER_STORY";
    FeatureProvidedSignals_StoryType[FeatureProvidedSignals_StoryType["OUR_STORY"] = 2] = "OUR_STORY";
    FeatureProvidedSignals_StoryType[FeatureProvidedSignals_StoryType["PUBLISHER_STORY"] = 3] = "PUBLISHER_STORY";
    FeatureProvidedSignals_StoryType[FeatureProvidedSignals_StoryType["FRIEND_STORY"] = 4] = "FRIEND_STORY";
    FeatureProvidedSignals_StoryType[FeatureProvidedSignals_StoryType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(FeatureProvidedSignals_StoryType || (FeatureProvidedSignals_StoryType = {}));
export var FeatureProvidedSignals_SpectaclesVersion;
(function (FeatureProvidedSignals_SpectaclesVersion) {
    FeatureProvidedSignals_SpectaclesVersion[FeatureProvidedSignals_SpectaclesVersion["UNKNOWN_SPECTACLES_VERSION"] = 0] = "UNKNOWN_SPECTACLES_VERSION";
    FeatureProvidedSignals_SpectaclesVersion[FeatureProvidedSignals_SpectaclesVersion["V1"] = 1] = "V1";
    FeatureProvidedSignals_SpectaclesVersion[FeatureProvidedSignals_SpectaclesVersion["V2"] = 2] = "V2";
    FeatureProvidedSignals_SpectaclesVersion[FeatureProvidedSignals_SpectaclesVersion["V3"] = 3] = "V3";
    FeatureProvidedSignals_SpectaclesVersion[FeatureProvidedSignals_SpectaclesVersion["V4"] = 4] = "V4";
    FeatureProvidedSignals_SpectaclesVersion[FeatureProvidedSignals_SpectaclesVersion["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(FeatureProvidedSignals_SpectaclesVersion || (FeatureProvidedSignals_SpectaclesVersion = {}));
function createBaseFeatureProvidedSignals() {
    return {
        mediaMetadata: undefined,
        storyMetadata: undefined,
        cameraSignals: undefined,
        snapSource: 0,
        contentManagerProvidedSignals: undefined,
        spectacleMetadata: undefined,
        boltSignals: undefined,
        lensesSignals: undefined,
        creativeToolsSignals: undefined,
        estimatedDurationForEventMs: 0,
        routingSignals: undefined,
        mdpMediaAttribution: undefined,
        size: 0,
        mediaSignals: undefined,
        perceptionSignals: undefined,
        operaSignals: undefined,
        recipientsSignals: undefined,
        billboardSignals: undefined,
        cognacSignals: undefined,
        hashSignals: undefined,
        edgeSignals: {},
        snapKitSignals: undefined,
        uploadSignals: undefined,
        ruids: [],
    };
}
export const FeatureProvidedSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.mediaMetadata !== undefined) {
            MediaMetadata.encode(message.mediaMetadata, writer.uint32(10).fork()).ldelim();
        }
        if (message.storyMetadata !== undefined) {
            FeatureProvidedSignals_StoryMetadata.encode(message.storyMetadata, writer.uint32(18).fork()).ldelim();
        }
        if (message.cameraSignals !== undefined) {
            CameraSignals.encode(message.cameraSignals, writer.uint32(26).fork()).ldelim();
        }
        if (message.snapSource !== 0) {
            writer.uint32(32).int32(message.snapSource);
        }
        if (message.contentManagerProvidedSignals !== undefined) {
            ContentManagerSignals.encode(message.contentManagerProvidedSignals, writer.uint32(42).fork()).ldelim();
        }
        if (message.spectacleMetadata !== undefined) {
            FeatureProvidedSignals_SpectacleMetadata.encode(message.spectacleMetadata, writer.uint32(50).fork()).ldelim();
        }
        if (message.boltSignals !== undefined) {
            BoltSignals.encode(message.boltSignals, writer.uint32(58).fork()).ldelim();
        }
        if (message.lensesSignals !== undefined) {
            LensesSignals.encode(message.lensesSignals, writer.uint32(66).fork()).ldelim();
        }
        if (message.creativeToolsSignals !== undefined) {
            CreativeToolsSignals.encode(message.creativeToolsSignals, writer.uint32(74).fork()).ldelim();
        }
        if (message.estimatedDurationForEventMs !== 0) {
            writer.uint32(80).uint32(message.estimatedDurationForEventMs);
        }
        if (message.routingSignals !== undefined) {
            RoutingSignals.encode(message.routingSignals, writer.uint32(90).fork()).ldelim();
        }
        if (message.mdpMediaAttribution !== undefined) {
            MdpMediaAttribution.encode(message.mdpMediaAttribution, writer.uint32(98).fork()).ldelim();
        }
        if (message.size !== 0) {
            writer.uint32(104).uint32(message.size);
        }
        if (message.mediaSignals !== undefined) {
            MediaSignals.encode(message.mediaSignals, writer.uint32(114).fork()).ldelim();
        }
        if (message.perceptionSignals !== undefined) {
            PerceptionSignals.encode(message.perceptionSignals, writer.uint32(122).fork()).ldelim();
        }
        if (message.operaSignals !== undefined) {
            OperaSignals.encode(message.operaSignals, writer.uint32(130).fork()).ldelim();
        }
        if (message.recipientsSignals !== undefined) {
            RecipientsSignals.encode(message.recipientsSignals, writer.uint32(138).fork()).ldelim();
        }
        if (message.billboardSignals !== undefined) {
            BillboardSignals.encode(message.billboardSignals, writer.uint32(146).fork()).ldelim();
        }
        if (message.cognacSignals !== undefined) {
            CognacSignals.encode(message.cognacSignals, writer.uint32(154).fork()).ldelim();
        }
        if (message.hashSignals !== undefined) {
            FeatureProvidedSignals_HashSignals.encode(message.hashSignals, writer.uint32(162).fork()).ldelim();
        }
        Object.entries(message.edgeSignals).forEach(([key, value]) => {
            FeatureProvidedSignals_EdgeSignalsEntry.encode({ key: key, value }, writer.uint32(170).fork()).ldelim();
        });
        if (message.snapKitSignals !== undefined) {
            SnapKitSignals.encode(message.snapKitSignals, writer.uint32(178).fork()).ldelim();
        }
        if (message.uploadSignals !== undefined) {
            UploadSignals.encode(message.uploadSignals, writer.uint32(186).fork()).ldelim();
        }
        for (const v of message.ruids) {
            Ruid.encode(v, writer.uint32(194).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFeatureProvidedSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.mediaMetadata = MediaMetadata.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.storyMetadata = FeatureProvidedSignals_StoryMetadata.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.cameraSignals = CameraSignals.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.snapSource = reader.int32();
                    break;
                case 5:
                    message.contentManagerProvidedSignals = ContentManagerSignals.decode(reader, reader.uint32());
                    break;
                case 6:
                    message.spectacleMetadata = FeatureProvidedSignals_SpectacleMetadata.decode(reader, reader.uint32());
                    break;
                case 7:
                    message.boltSignals = BoltSignals.decode(reader, reader.uint32());
                    break;
                case 8:
                    message.lensesSignals = LensesSignals.decode(reader, reader.uint32());
                    break;
                case 9:
                    message.creativeToolsSignals = CreativeToolsSignals.decode(reader, reader.uint32());
                    break;
                case 10:
                    message.estimatedDurationForEventMs = reader.uint32();
                    break;
                case 11:
                    message.routingSignals = RoutingSignals.decode(reader, reader.uint32());
                    break;
                case 12:
                    message.mdpMediaAttribution = MdpMediaAttribution.decode(reader, reader.uint32());
                    break;
                case 13:
                    message.size = reader.uint32();
                    break;
                case 14:
                    message.mediaSignals = MediaSignals.decode(reader, reader.uint32());
                    break;
                case 15:
                    message.perceptionSignals = PerceptionSignals.decode(reader, reader.uint32());
                    break;
                case 16:
                    message.operaSignals = OperaSignals.decode(reader, reader.uint32());
                    break;
                case 17:
                    message.recipientsSignals = RecipientsSignals.decode(reader, reader.uint32());
                    break;
                case 18:
                    message.billboardSignals = BillboardSignals.decode(reader, reader.uint32());
                    break;
                case 19:
                    message.cognacSignals = CognacSignals.decode(reader, reader.uint32());
                    break;
                case 20:
                    message.hashSignals = FeatureProvidedSignals_HashSignals.decode(reader, reader.uint32());
                    break;
                case 21:
                    const entry21 = FeatureProvidedSignals_EdgeSignalsEntry.decode(reader, reader.uint32());
                    if (entry21.value !== undefined) {
                        message.edgeSignals[entry21.key] = entry21.value;
                    }
                    break;
                case 22:
                    message.snapKitSignals = SnapKitSignals.decode(reader, reader.uint32());
                    break;
                case 23:
                    message.uploadSignals = UploadSignals.decode(reader, reader.uint32());
                    break;
                case 24:
                    message.ruids.push(Ruid.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e;
        const message = createBaseFeatureProvidedSignals();
        message.mediaMetadata =
            object.mediaMetadata !== undefined && object.mediaMetadata !== null
                ? MediaMetadata.fromPartial(object.mediaMetadata)
                : undefined;
        message.storyMetadata =
            object.storyMetadata !== undefined && object.storyMetadata !== null
                ? FeatureProvidedSignals_StoryMetadata.fromPartial(object.storyMetadata)
                : undefined;
        message.cameraSignals =
            object.cameraSignals !== undefined && object.cameraSignals !== null
                ? CameraSignals.fromPartial(object.cameraSignals)
                : undefined;
        message.snapSource = (_a = object.snapSource) !== null && _a !== void 0 ? _a : 0;
        message.contentManagerProvidedSignals =
            object.contentManagerProvidedSignals !== undefined && object.contentManagerProvidedSignals !== null
                ? ContentManagerSignals.fromPartial(object.contentManagerProvidedSignals)
                : undefined;
        message.spectacleMetadata =
            object.spectacleMetadata !== undefined && object.spectacleMetadata !== null
                ? FeatureProvidedSignals_SpectacleMetadata.fromPartial(object.spectacleMetadata)
                : undefined;
        message.boltSignals =
            object.boltSignals !== undefined && object.boltSignals !== null
                ? BoltSignals.fromPartial(object.boltSignals)
                : undefined;
        message.lensesSignals =
            object.lensesSignals !== undefined && object.lensesSignals !== null
                ? LensesSignals.fromPartial(object.lensesSignals)
                : undefined;
        message.creativeToolsSignals =
            object.creativeToolsSignals !== undefined && object.creativeToolsSignals !== null
                ? CreativeToolsSignals.fromPartial(object.creativeToolsSignals)
                : undefined;
        message.estimatedDurationForEventMs = (_b = object.estimatedDurationForEventMs) !== null && _b !== void 0 ? _b : 0;
        message.routingSignals =
            object.routingSignals !== undefined && object.routingSignals !== null
                ? RoutingSignals.fromPartial(object.routingSignals)
                : undefined;
        message.mdpMediaAttribution =
            object.mdpMediaAttribution !== undefined && object.mdpMediaAttribution !== null
                ? MdpMediaAttribution.fromPartial(object.mdpMediaAttribution)
                : undefined;
        message.size = (_c = object.size) !== null && _c !== void 0 ? _c : 0;
        message.mediaSignals =
            object.mediaSignals !== undefined && object.mediaSignals !== null
                ? MediaSignals.fromPartial(object.mediaSignals)
                : undefined;
        message.perceptionSignals =
            object.perceptionSignals !== undefined && object.perceptionSignals !== null
                ? PerceptionSignals.fromPartial(object.perceptionSignals)
                : undefined;
        message.operaSignals =
            object.operaSignals !== undefined && object.operaSignals !== null
                ? OperaSignals.fromPartial(object.operaSignals)
                : undefined;
        message.recipientsSignals =
            object.recipientsSignals !== undefined && object.recipientsSignals !== null
                ? RecipientsSignals.fromPartial(object.recipientsSignals)
                : undefined;
        message.billboardSignals =
            object.billboardSignals !== undefined && object.billboardSignals !== null
                ? BillboardSignals.fromPartial(object.billboardSignals)
                : undefined;
        message.cognacSignals =
            object.cognacSignals !== undefined && object.cognacSignals !== null
                ? CognacSignals.fromPartial(object.cognacSignals)
                : undefined;
        message.hashSignals =
            object.hashSignals !== undefined && object.hashSignals !== null
                ? FeatureProvidedSignals_HashSignals.fromPartial(object.hashSignals)
                : undefined;
        message.edgeSignals = Object.entries((_d = object.edgeSignals) !== null && _d !== void 0 ? _d : {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[Number(key)] = Value.fromPartial(value);
            }
            return acc;
        }, {});
        message.snapKitSignals =
            object.snapKitSignals !== undefined && object.snapKitSignals !== null
                ? SnapKitSignals.fromPartial(object.snapKitSignals)
                : undefined;
        message.uploadSignals =
            object.uploadSignals !== undefined && object.uploadSignals !== null
                ? UploadSignals.fromPartial(object.uploadSignals)
                : undefined;
        message.ruids = ((_e = object.ruids) === null || _e === void 0 ? void 0 : _e.map((e) => Ruid.fromPartial(e))) || [];
        return message;
    },
};
function createBaseFeatureProvidedSignals_StoryMetadata() {
    return { isPublicStory: false, isOfficialStory: false, storyType: 0 };
}
export const FeatureProvidedSignals_StoryMetadata = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isPublicStory === true) {
            writer.uint32(8).bool(message.isPublicStory);
        }
        if (message.isOfficialStory === true) {
            writer.uint32(16).bool(message.isOfficialStory);
        }
        if (message.storyType !== 0) {
            writer.uint32(24).int32(message.storyType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFeatureProvidedSignals_StoryMetadata();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.isPublicStory = reader.bool();
                    break;
                case 2:
                    message.isOfficialStory = reader.bool();
                    break;
                case 3:
                    message.storyType = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseFeatureProvidedSignals_StoryMetadata();
        message.isPublicStory = (_a = object.isPublicStory) !== null && _a !== void 0 ? _a : false;
        message.isOfficialStory = (_b = object.isOfficialStory) !== null && _b !== void 0 ? _b : false;
        message.storyType = (_c = object.storyType) !== null && _c !== void 0 ? _c : 0;
        return message;
    },
};
function createBaseFeatureProvidedSignals_SpectacleMetadata() {
    return { isSpectacle: false, spectaclesVersion: 0 };
}
export const FeatureProvidedSignals_SpectacleMetadata = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isSpectacle === true) {
            writer.uint32(8).bool(message.isSpectacle);
        }
        if (message.spectaclesVersion !== 0) {
            writer.uint32(16).int32(message.spectaclesVersion);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFeatureProvidedSignals_SpectacleMetadata();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.isSpectacle = reader.bool();
                    break;
                case 2:
                    message.spectaclesVersion = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseFeatureProvidedSignals_SpectacleMetadata();
        message.isSpectacle = (_a = object.isSpectacle) !== null && _a !== void 0 ? _a : false;
        message.spectaclesVersion = (_b = object.spectaclesVersion) !== null && _b !== void 0 ? _b : 0;
        return message;
    },
};
function createBaseFeatureProvidedSignals_HashSignals() {
    return { userId: new Uint8Array(), seed: new Uint8Array(), namespace: 0 };
}
export const FeatureProvidedSignals_HashSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.userId.length !== 0) {
            writer.uint32(10).bytes(message.userId);
        }
        if (message.seed.length !== 0) {
            writer.uint32(18).bytes(message.seed);
        }
        if (message.namespace !== 0) {
            writer.uint32(24).int32(message.namespace);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFeatureProvidedSignals_HashSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.userId = reader.bytes();
                    break;
                case 2:
                    message.seed = reader.bytes();
                    break;
                case 3:
                    message.namespace = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseFeatureProvidedSignals_HashSignals();
        message.userId = (_a = object.userId) !== null && _a !== void 0 ? _a : new Uint8Array();
        message.seed = (_b = object.seed) !== null && _b !== void 0 ? _b : new Uint8Array();
        message.namespace = (_c = object.namespace) !== null && _c !== void 0 ? _c : 0;
        return message;
    },
};
function createBaseFeatureProvidedSignals_EdgeSignalsEntry() {
    return { key: 0, value: undefined };
}
export const FeatureProvidedSignals_EdgeSignalsEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== 0) {
            writer.uint32(8).int32(message.key);
        }
        if (message.value !== undefined) {
            Value.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFeatureProvidedSignals_EdgeSignalsEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.key = reader.int32();
                    break;
                case 2:
                    message.value = Value.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseFeatureProvidedSignals_EdgeSignalsEntry();
        message.key = (_a = object.key) !== null && _a !== void 0 ? _a : 0;
        message.value = object.value !== undefined && object.value !== null ? Value.fromPartial(object.value) : undefined;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
