// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/media_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.cdp.cof";
export var MediaType;
(function (MediaType) {
    MediaType[MediaType["MEDIA_TYPE_UNSET"] = 0] = "MEDIA_TYPE_UNSET";
    MediaType[MediaType["IMAGE"] = 1] = "IMAGE";
    MediaType[MediaType["VIDEO"] = 2] = "VIDEO";
    MediaType[MediaType["GIF"] = 3] = "GIF";
    MediaType[MediaType["AUDIO"] = 4] = "AUDIO";
    MediaType[MediaType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MediaType || (MediaType = {}));
function createBaseMediaSignals() {
    return { mediaDurationMs: 0, mediaType: 0 };
}
export const MediaSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.mediaDurationMs !== 0) {
            writer.uint32(8).uint32(message.mediaDurationMs);
        }
        if (message.mediaType !== 0) {
            writer.uint32(16).int32(message.mediaType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMediaSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.mediaDurationMs = reader.uint32();
                    break;
                case 2:
                    message.mediaType = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMediaSignals();
        message.mediaDurationMs = (_a = object.mediaDurationMs) !== null && _a !== void 0 ? _a : 0;
        message.mediaType = (_b = object.mediaType) !== null && _b !== void 0 ? _b : 0;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
