// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/opera_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.cdp.cof";
function createBaseOperaSignals() {
    return { contentViewSource: 0, playbackItemType: 0, useVerticalNavigation: false };
}
export const OperaSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.contentViewSource !== 0) {
            writer.uint32(8).int32(message.contentViewSource);
        }
        if (message.playbackItemType !== 0) {
            writer.uint32(16).int32(message.playbackItemType);
        }
        if (message.useVerticalNavigation === true) {
            writer.uint32(24).bool(message.useVerticalNavigation);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOperaSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.contentViewSource = reader.int32();
                    break;
                case 2:
                    message.playbackItemType = reader.int32();
                    break;
                case 3:
                    message.useVerticalNavigation = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseOperaSignals();
        message.contentViewSource = (_a = object.contentViewSource) !== null && _a !== void 0 ? _a : 0;
        message.playbackItemType = (_b = object.playbackItemType) !== null && _b !== void 0 ? _b : 0;
        message.useVerticalNavigation = (_c = object.useVerticalNavigation) !== null && _c !== void 0 ? _c : false;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
