// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/perception_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.cdp.cof";
function createBasePerceptionSignals() {
    return { scannedCategoryIds: [] };
}
export const PerceptionSignals = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.scannedCategoryIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePerceptionSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.scannedCategoryIds.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBasePerceptionSignals();
        message.scannedCategoryIds = ((_a = object.scannedCategoryIds) === null || _a === void 0 ? void 0 : _a.map((e) => e)) || [];
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
