// @snap/camera-kit/dist/generated-proto/pb_schema/cdp/cof/upload_provided_signals.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.cdp.cof";
export var ChunkUploadPreference;
(function (ChunkUploadPreference) {
    ChunkUploadPreference[ChunkUploadPreference["PREFERENCE_UNSET"] = 0] = "PREFERENCE_UNSET";
    ChunkUploadPreference[ChunkUploadPreference["GCS"] = 1] = "GCS";
    ChunkUploadPreference[ChunkUploadPreference["S3"] = 2] = "S3";
    ChunkUploadPreference[ChunkUploadPreference["AUTO"] = 3] = "AUTO";
    ChunkUploadPreference[ChunkUploadPreference["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ChunkUploadPreference || (ChunkUploadPreference = {}));
function createBaseUploadSignals() {
    return { chunkUploadSupportRequired: false, cloudfrontPop: "", gcdnPop: "", chunkUploadPreference: 0 };
}
export const UploadSignals = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.chunkUploadSupportRequired === true) {
            writer.uint32(8).bool(message.chunkUploadSupportRequired);
        }
        if (message.cloudfrontPop !== "") {
            writer.uint32(18).string(message.cloudfrontPop);
        }
        if (message.gcdnPop !== "") {
            writer.uint32(26).string(message.gcdnPop);
        }
        if (message.chunkUploadPreference !== 0) {
            writer.uint32(32).int32(message.chunkUploadPreference);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUploadSignals();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.chunkUploadSupportRequired = reader.bool();
                    break;
                case 2:
                    message.cloudfrontPop = reader.string();
                    break;
                case 3:
                    message.gcdnPop = reader.string();
                    break;
                case 4:
                    message.chunkUploadPreference = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseUploadSignals();
        message.chunkUploadSupportRequired = (_a = object.chunkUploadSupportRequired) !== null && _a !== void 0 ? _a : false;
        message.cloudfrontPop = (_b = object.cloudfrontPop) !== null && _b !== void 0 ? _b : "";
        message.gcdnPop = (_c = object.gcdnPop) !== null && _c !== void 0 ? _c : "";
        message.chunkUploadPreference = (_d = object.chunkUploadPreference) !== null && _d !== void 0 ? _d : 0;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
