// @snap/camera-kit/dist/generated-proto/pb_schema/snapdoc/capture_characteristics.js
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "snapchat.snapdoc";
function createBaseCaptureCharacteristics() {
    return { flashOn: false, frontFacing: false };
}
export const CaptureCharacteristics = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.flashOn === true) {
            writer.uint32(8).bool(message.flashOn);
        }
        if (message.frontFacing === true) {
            writer.uint32(16).bool(message.frontFacing);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCaptureCharacteristics();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.flashOn = reader.bool();
                    break;
                case 2:
                    message.frontFacing = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseCaptureCharacteristics();
        message.flashOn = (_a = object.flashOn) !== null && _a !== void 0 ? _a : false;
        message.frontFacing = (_b = object.frontFacing) !== null && _b !== void 0 ? _b : false;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
